lib/Elastica/Index.php

```diff
@@ -295,7 +293,7 @@ public function exists()
      */
     public function createSearch($query = '', $options = null)
     {
-        $search = new Search($this->getClient());
+        $search = new Search($this->getClient(), $this->getClient()->getResultSetBuilder());
```

Comment:
As far as I understand, every Search can be done with a different Result Builder, but by default is the one used set in the client. So instead of passing it here, we could make Search fetching the default result builder from client if the param as null.